### ISSUE TITLE
Create CloudFront distribution for harmonized domain

### ISSUE THREAD

**Issue by alukach** [NASA-IMPACT/veda-backend#123]
In order to serve our VEDA services from a "pretty-url" (ie `veda.{some-org}.nasa.gov`), we need to be able to provide the domain administrators with a CloudFront distribution to which the pretty-url will send traffic.

It likely makes sense to generate the CloudFront distribution with a separate repo and then to begin adding routes to the repo to use for routing traffic to various services (ie utilizing CloudFront as a reverse proxy).

**Comment by alukach** [comment]
We did this for CSDA.  Afterwards, I wrote up my steps in [an internal DS ticket](https://github.com/developmentseed/how/issues/270) (which I cross-posted to [my own website](https://alukach.com/posts/using-cloudfont-as-a-reverse-proxy/) for public consumption).  It contains some code samples that may be of help if it's a goal to do this via CDK.

**Comment by xhagrg** [comment]
@freitagb this is what you were looking for!

**Comment by slesaad** [comment]
that's awesome, thanks @alukach, the CDK code will definitely be helpful! :)

**Comment by slesaad** [comment]
I went a step further and assigned the `delta-backend.xyz` domain (that already existed, but was not being used) to the cloudfront distribution to test the behavior.

Some updates I had to make:
- create and add a SSL certificate to the distribution
- add the alternate domain name in the distribution that's the same as the domain name in route53
- in route53, i added an A record (alias) to delta-backend.xyz, and pointed it to the cloudfront distribution

So now this works:
[delta-backend.xyz](delta-backend.xyz) -  points to the STAC browser
[delta-backend.xyzapi/stac](delta-backend.xyzapi/stac) -  points to the STAC API
[delta-backend.xyzapi/raster](delta-backend.xyzapi/raster) -  points to the raster API
[delta-backend.xyzapi/ingest](delta-backend.xyzapi/ingest) -  points to the STAC ingestor API

**Comment by anayeaye** [comment]
@slesaad we may also need to open a issues to look into what changes will be needed for the stac browser and raster APIs to work with the extended path prefix on the STAC API. According to the stac-api spec the additional path prefix is allowed but some of our tools expect a the root of the STAC catalog to be a base url.

The browser is not a requirement for VEDA but it looks like some work will be needed to work with the additional `/api/stac/` prefix in the path. For this, however, we could probably deploy our static browsers using the API gateway URL.
https://radiantearth.github.io/stac-browser/#/external/https://delta-backend.xyz/api/stac

The raster API swagger (I haven't had time to try out any API requests but this looks like we'll need to make some adjustments for the open api docs which the current configuration expects to be at the root)
https://delta-backend.xyz/api/raster/docs

**Comment by slesaad** [comment]
@anayeaye that's some really good points. I'll create that ticket!

**Comment by anayeaye** [comment]
@slesaad side note about the raster API: we were investigating the validation errors in dev and found that the lambda was no longer able to assume the data access role with this error
```
Unable to assume role arn:aws:iam::114506680961:role/veda-data-store-read-staging with exception=An error occurred (AccessDenied) when calling the AssumeRole operation: User: arn:aws:sts::853558080719:assumed-role/veda-backend-uah-dev-rasterapilambdaServiceRoleE47-187OLESEZ31PM/veda-backend-uah-dev-rasterapilambda27FDBBDF-mGHHmWBvJUKi is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::114506680961:role/veda-data-store-read-staging
```

I don't see why the additional cloudfront would matter for the existing dev endpoint called from dev-raster.delta-backend.com but on a whim I updated the conditions in the MCP data access trust policy to allow delta-backed instead of delta-backend-staging and the assume role error is no longer occurring (which makes no sense to me but the error did just start on the 5th--I searched back a couple months in the logs). Unfortunately the raster API including the cog viewer is still not working as expected. There are enough simultaneous changes in the dev stack right now that the raster API behavior may be unrelated to the new cloudfront distribution. 

<img width="400" alt="mcp-data-access-policy-conditions" src="https://user-images.githubusercontent.com/3386265/211084439-d76c97f8-a659-41c9-b6c8-f802201a2d02.png">

cc: @smohiudd
